### PR TITLE
Added status parameter to /etl-status endpoint that determines data returned

### PR DESCRIPTION
**Pull request by that-guy-is-it**

### CODE REVIEW

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java [hunk 1]
```diff
@@ -55,7 +56,7 @@ public void clearResourceStatusScheduler() {
     }
   }
 
-  ResponseEntity<Health> resourceStatusHealth(Instant now) {
+  ResponseEntity<Health> resourceStatusHealth(Instant now, boolean shouldReturnStatus) {
     hasCachedResourceStatus.set(true);
     Instant tooLongAgo = now.minus(36, ChronoUnit.HOURS);
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java:61]: Nothing relating to etl status should occur unless the boolean is true.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java [hunk 2]
```diff
@@ -85,7 +92,8 @@ public void clearResourceStatusScheduler() {
    */
   @Cacheable("resource-status")
   @GetMapping
-  public ResponseEntity<Health> resourceStatusHealth() {
-    return resourceStatusHealth(Instant.now());
+  public ResponseEntity<Health> resourceStatusHealth(
+      @RequestParam(defaultValue = "true", required = false) boolean status) {
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java:96]: Don't forget to use `value` to tell the annotation what the queryParam is.

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java [hunk 3]
```diff
@@ -64,17 +65,23 @@ public void clearResourceStatusScheduler() {
...
     if (!overallHealth.getStatus().equals(Status.UP)) {
       return ResponseEntity.status(HttpStatus.SERVICE_UNAVAILABLE).body(overallHealth);
     }
     return ResponseEntity.ok(overallHealth);
   }
 
+  ResponseEntity<Health> resourceStatusHealth(Instant now) {
+    return resourceStatusHealth(now, true);
+  }
+
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java:84]: Is this guy doing anything other than being used in tests?

Comment (jblefkove-va) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java:84]: Yeah you don't need this. The tests can specify both arguments.

Comment (that-guy-is-it) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java:84]: Hmm, yes its only used in tests. I added it as a way to allow for a default value in the logic for testing. Is there a better approach?

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java [hunk 4]
```diff
@@ -64,17 +65,23 @@ public void clearResourceStatusScheduler() {
     boolean isDown =
         statusDetails.isEmpty()
             || statusDetails.stream().anyMatch(h -> h.getStatus().equals(Status.DOWN));
-    Health overallHealth =
+    Health.Builder health =
         Health.status(new Status(isDown ? "DOWN" : "UP", "Resource ETL Statuses"))
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java:69]: Name should be "Overall Status"

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java [hunk 5]
```diff
@@ -64,17 +65,23 @@ public void clearResourceStatusScheduler() {
...
-            .withDetail("time", now)
-            .withDetail("statuses", statusDetails)
-            .build();
+            .withDetail("time", now);
+    if (shouldReturnStatus) {
+      health.withDetail("statuses", statusDetails);
```
Comment (jhulbert-va) [inline data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java:72]: "Resource ETL Statuses"

**Review by bschofield-va — CHANGES_REQUESTED** [review]
I'm blocking merge until I hear back from Alex. I have some outstanding questions that could impact this effort

File: data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java (hunk 3 above)
Comment (that-guy-is-it) [reply data-query/src/main/java/gov/va/api/health/dataquery/service/controller/etlstatus/LatestResourceEtlStatusController.java:84]: Alright, I will force it in tests then.

**Comment by jhulbert-va** [comment]
> I'm blocking merge until I hear back from Alex. I have some outstanding questions that could impact this effort

This PR is just updating the already existing ETL endpoint. Are you thinking of [API-2685](https://vajira.max.gov/browse/API-2685)?

**Pull request closed without merging**